scripts/upload_dataset.py

```diff
@@ -0,0 +1,193 @@
+        tags = self.trainer.get_tags(self.project.id)
+        for tag in tags:
+            if tag.name == tag_name:
+                return self.trainer.get_tag(self.project.id, tag.id)
+
+        return self.trainer.create_tag(self.project.id, tag_name)
+
+    def _read_and_resize_image(self, image_path: str) -> bytes:
+        image = cv2.imread(image_path, cv2.IMREAD_UNCHANGED)
+        current_byte_size = len(cv2.imencode(".jpg", image)[1].tobytes())
+        if current_byte_size > self.max_byte_size:
+            scale_factor: int = int(cbrt(self.max_byte_size / current_byte_size) * 95)
+            encode_param = [int(cv2.IMWRITE_JPEG_QUALITY), scale_factor]
+            image_bytes = cv2.imencode(".jpg", image, encode_param)[1].tobytes()
+        else:
+            image_bytes = cv2.imencode(".jpg", image)[1].tobytes()
+
+        return image_bytes
+
+    def _read_annotation_file(self, annotation_path: str) -> list:
+        annotations = []
+        with open(annotation_path, "r") as f:
+
+            for line in f:
+                line = line.strip()
+                parameter_list = line.split(" ")
+                label = int(parameter_list[0])
+                x, y, w, h = list(map(float, parameter_list[1:]))
+
+                left = x - w / 2
+                if left < 0:  # Accounting for previous rounding error
+                    left = 0
+                top = y - h / 2
+                if top < 0:  # Accounting for previous rounding error
+                    top = 0
+
+                if left + w > 1:  # Accounting for previous rounding error
+                    w = 1 - left
+                if top + h > 1:  # Accounting for previous rounding error
+                    h = 1 - top
+
+                if label == 0:
+                    tag_id = self.red_car_seal_tag.id
+                elif label == 1:
+                    tag_id = self.green_car_seal_tag.id
+                else:
+                    raise ValueError(f"Wrong label {label} at {annotation_path}")
+                annotations.append(
+                    Region(
+                        tag_id=tag_id,
+                        left=left,
+                        top=top,
+                        width=w,
+                        height=h,
+                    )
+                )
+        return annotations
+
+    def main(self) -> None:
+        self.project: Project = self._connect_to_or_create_project(
+            project_name=self.project_name
+        )
+        # Make two tags in the new project
+        self.green_car_seal_tag = self._get_or_create_tag("green_car_seal")
+        self.red_car_seal_tag = self._get_or_create_tag("red_car_seal")
+
+        existing_image_count = self.trainer.get_image_count(project_id=self.project.id)
+
+        dataset_path = os.path.join(os.path.dirname(__file__), "../dataset")
+
+        file_number = 0
+        for file_name in txt_file_paths:
+            if file_number < existing_image_count:
```

Comment:
Slice the `self.files_to_upload` list and set `file_number` to `existing_image_count` before starting the iteration instead?